### ISSUE TITLE
How TinyWall should have been....  but there are some Issues.

### ISSUE THREAD

**Issue by leeuniverse** [wokhansoft/WFN#73]
I have been looking for this program for YEARS...  When TinyWall came out, I had hoped it would do almost all that this program does, so I'm happy someone has finally done it and I finally found it.

It's messed up too that this program has been here for so long and I never found it, even though about once a year I do a MASSIVE Google search for Free Firewalls, especially Windows Firewall based ones with extra functionality (cause I've been wanting this forever), and this one never came up.  However, this time around in my search at least one of the places that list free firewalls and alternatives to TinyWall, this program finally popped up, and so did some others.

Anyway, I first have some recommendations to get the word out about this program.  Marketing has been your problem, as well as some bugs which I'll get into below.
You first need to have an actual "Website"...  I don't know if Github gives you at least a Subdomain, but it's easy and free to get a free Responsive HTML Template (or WordPress etc.), and use a free web host to start with.  I recommend InfinityFree, they actually allow you to use your site/have some traffic and lots of features before trying to ban you/forcing you to upgrade, and when you do their upgrade is cheap with their parent provider.

Another thing you can do, is I'm guessing a lot of these places that recommend Programs etc., you can likely submit yours for review, and to download places etc. for download.  Have YouTube videos advertising your program, explaining features and how to do things.  There are so many EASY things you can do.  What I would recommend is do a major update of your program, fixing all the major bugs, and then do these things, and your program WILL start to take off once people actually know it exists.  I mean, I just found you, and I'm a power user in searching, that is how BAD your marketing is.

Now to the issues I have, some of which are likely going to make me use either Free Firewall, TinyWall, or SimpleWall (still testing all these) etc. instead of WFN atm.

1. Likely an easy fix, we need column sizes/positions, and the program window to remain where you left them when you tab between the pages of the app, and when you've opened and closed the app.

2. The Minimize to System Tray setting isn't working.  Though, I saw we don't have to keep the program open for it to work, so I guess that's not a huge issue, though it should also be an easy fix.

3. Programs do not remember they have been allowed...  Now, I don't know if it's "how" I'm allowing them or what, you have so many options in the pop-up with ports etc. I'm confused, cause the default Allow wasn't working, so I started selecting ALL of the options, and I "think" they might have calmed down some, I'm speaking especially for my Anti-Virus program "360 Total Security".  There is one particular program it runs "qhactivedefense.exe" in which I've had to create rules for it dozens of times.  Now, I think it "might" have calmed down, but we'll see.  But, it's not the only program I've had to create more than one rule for the same EXE.

Two more examples of this are your own program WFN, and MS Outlook Mail.  First, your program for some reason when I opened it up once it started going crazy forcing me to click allow a million times (I think it was when I went to the Map section of the program after opening it).  To stop it I had to close the program and then eventually I got through all the Notifications allowing them all.  The same exact thing happened with Outlook, though I don't think I had opened it yet, it occurred on it's first run, again had to close it to get through the notifications.

In this vein, we clearly need a Tutorial on how to use the Notifications to maybe prevent some of this I'm assuming, since like I mentioned when I selected ALL the options, the notifications have seemed to calm down now.

4. See my EXE extractor post to answer a question you've been trying to figure out related to "Installing" the program.

5. We need to be able to "Edit" the Rules...  I think an easy way you can do this is to just allow each Row (the specific Column in that row) to be editable when like maybe on a right-click.  This should likely be an easy way to implement the editability of a rule.

6. Have a setting in which ALL Windows OS related activity is automatically allowed.  So, that way we just need to create rules for 3rd Party Applications.  Also in that note, have a separate window (or separate Tabs likely better) for Connections, Firewall Rules, and Security Log which show ONLY MS System connections, rules, etc.  This can help better compartmentalize the activities of our system and applications to see what's going on.

Off the top of my head I think that's it...
Anyway, even if I don't continue using your program atm due to the issues I've mentioned, I plan on coming back, because yours is the ONLY one having everything pretty much that I'm looking for.  Hope you can fix these things soon, provide more instructions, market things more etc. so we can make this program how it should be (and how Windows Firewall should have been all along if competent people worked at Microsoft).

Thanks....  Lee

p.s. I'm actually willing to create your website for you for FREE...  I won't do anything complex, just the essentials.  But you can see here for some of what I can do http://www.evergrowbusiness.org (still updating the projects section, finally found screenshots of some older nicer site designs I had worked on in the past, so will be putting those up likely by tomorrow.)  :)  (oh, and find me on Discord if you want to chat)

**Comment by leeuniverse** [comment]
Welp...  Yep, I can't use this program unfortunately until the above serious issues are fixed.

I just tried to run "Pandora" and I literally clicked Allow, and the different settings a 100+ ways, starting the program over and over again and IT WOULD NOT WORK...

So, sadly I HAVE to remove WFN and try something else.  :(

**Comment by wokhan** [comment]
Hi (again 😁), thank you for this complete message.

First, regarding marketing : something that doesnt exist cannot be called bad... So ours is not bad but only not a priority or even something we considered. WFN started as a private project and has been published on huge sites (major geeks, download.com,....) back then. It had been downloaded almost 180 000 times on all those sources which I wouldn't have predicted in any way, then people asked for an open source version... Here we are, almost 10 years later.
We have to time for advertising a product that is in fact still in beta (as you were able to see 😉), since we already don't have time to fix some well known bugs.
What we need yet would be contributors, maybe "marketing" is the way to go for this also, but still I cannot invest time on this...

Regarding the website, I updated mine (https://www.wokhan.com) but it doesn't contain much. This is also a hobby I don't have time for anymore.

If something had to be prioritized appart from the bug fixes, it would be the documentation which is indeed inexistant. Time consuming, as well... Feel free to help on that one 😉

Regarding the bugs : 
1. Columns and window size / position : will be added to our improvements list
2. Minimize to tray should work, looks like a regression... To be investigated.
3. Program re-raising notifications while they have been allowed : this is of course not expected if the rule has been created properly, which is what you did. Some programs are known to cause multiple notifications but Outlook and the like shouldn't.
4. Auro extracting archive instead of manual unzipping : answered in the corresponding issue thread. 
5. Rules editability : to be added in our improvements list
6. Already asked for and I think it is on our list, but still not implemented... 

Thank you again and again: this is a community project and you're more than welcome if you want to work on it (could be static html pages for the documentation, hosted here at github) 😉

**Comment by leeuniverse** [comment]
Thanks for the response...  :)

1. Ya, it doesn't take a lot to market more that would massively boost your reach.  Other than the things I've already mentioned, as one example, I saw your Website and it's an attractive design, however, with just one flaw, it needs MORE words describing WFN so the search engines can find it better, aka better SEO.  I mean, you almost have nothing there, so that could be improved fairly easily.

2. See, and you prove my point, you've been out there 10 years and I simply didn't see you and if I did I must have passed by it, I'm guessing you were "paid" at first?

3. Ah, that's too bad you don't have time for your hobby....  LOL  :)
Hmmm...  Documentation?  That might be a bit harder, not that I couldn't do it I could (I've created a simple CHM one for my mod pack I develed for a game that I included in my Inno Installer for it), but it's only harder in that I don't really know the program, and I'm not the one that get's the questions and issues who can then make an FAQ, and I'm not going to be using the program atm so that makes it harder etc.  But, I will mul it over look over things more.  Have you created a file with peoples questions, issues and your answers at all, or would we have to go through this section and pick out things?

4. (1)Great...  (2) Great....  (3) BTW it was the App version of Outlook which comes with Windows 10.  I mentioned Pandora doing the same to where I wasn't even able to use it, also an MS Store App.  But, then you have my installed programs which did similar, though they eventually called down, though I did do the same exact settings multiple times for the 360 Total Security, and I think a couple of others though they only went a couple of times.  Remember also WFN had it happen too when I went to the Map Tab of the program (when I had been there already several times before).
(4) Sounds good, people just sometimes like the simplicity of an EXE install, so was just throwing that out there, maybe it could be a secondary download for those who like it, that's what a lot of Developers do.  (5) Great...  (6) Great...

5. Sounds good, I'll look things over further, and when I have more time, I'm jammed atm preparing different things for me to look for work etc. so just been updating various things, long hours.  But, I definitely want your program to succeed....  I've found one almost just like yours with slightly less features that's been working great just like yours should "Simplewall"...., though I'm still going to try "Free Firewall" which doesn't use Windows Firewall to see which I like best as more simple doing what's needed firewalls.

Anyway, I look forward to eventually using WFN as my in love program, hoping you can have time to fix the bugs at least, and looking forward to also helping you.  BTW, you should release the latest version, just so we can have the latest fixes to test the program etc. which has been like two years since the last release.

Take care, Lee  👍

**Comment by leeuniverse** [comment]
Hey, so I tested "Free Firewall" and it's somewhat okay, but I think it's actually doing the SAME thing with some programs as yours with the "million notifications" BUG for the same connection/app, but what they do is their Notification Window it actually has a "scrollbar" which displays the million notifications LOL, and at the bottom, they have Accept All or Block All....

So, I guess that's one way to deal with a bug with their program.  LOL
So, not going with that program.  I've went back to SimpleWall which just works nice... at least until the day WFN is working properly, then I'll like it better.  :)

Take care...  See ya in a while when things are settled again for me.

**Comment by wokhan** [comment]
Hi,
Thanks for your response to my answer 😉

1. My website is indeed almost empty, mainly because it's for "fun" (I like to update it every couple of years or so... keeping it while it's not that useful, but it doesn't cost me anything so...).

2. I've never been paid, except for a few donations which clearly don't match the time invested at first in this project. But I haven't created this tool for money anyway (or it would have been a paid solution in MS store for years). Open sourcing was a theoretical way to involve more people when it was more visible on security websites (it has been) ; it failed. Apart from @AtlasHackert and @harrwiss, whom I thank for their involvement, developers are not really interested in the project.

3. Yeah my actual job, my baby, my house, etc... don't leave me that much free time, unfortunately... Maybe WFN will be my top priority again when I'm retired 😁😉 And we haven't kept a proper FAQ or alike to serve as a base for documentation... So it would indeed be a gathering of information from github issues or emails I got all those years...

4. (+ free firewall comment above) This is probably the highest priority issue then, as it looks like it's a regression of sorts. It wasn't that bad with older versions...

5. Thank you, I see WFN still can raise some interest and it makes me at the same time proud of what has been achieved... and ashamed that it didn't go far enough. We are so close from a finished product, but it has been that way for years now... It's not dead, it's not that alive either... 😉

Regarding devs, the main issue here is that I cannot help with a knowledge transfer yet, so we need autonomous clever skilled .Net devs.... Ready to dig in the code on their own, at least at first...

To be continued!

Take care

**Comment by harrwiss** [comment]
Hi all, seems we're all in the same boat wanting but lacking time to do all we want 😉 

Since we haven't had a new beta release since March 2018 I created one in my fork for download. You can find the download together with the list of enhancements since then here: [WFN Version History and Download](https://github.com/harrwiss/WFN/wiki)

@leeuniverse: Maybe you could give it a try and see whether it solves some of your issues and give us some feedback. In addition I also created a FAQ page based on your questions above as a start: [WFN FAQ](https://github.com/harrwiss/WFN/wiki/FAQ)

Cheers.

**Comment by wokhan** [comment]
Hi @harrwiss, I guess it's time for you to join the main repository.
Forks are sometimes great, but here I'm afraid it could cause confusion for some people :-)
Would you be ok with this? Would mean you'll have to create Merge Requests instead of Pull Requests (@AtlasHackert or I will review them, if it's ok for both of you 🙂), the goal being to keep a single entrypoint for users wanting to DL the actual latest version 😉

**Comment by harrwiss** [comment]
Hi @wokhan, think your suggestion makes absolutely sense! However, let me say that I have some concerns about what I could get in return for what looks like additional effort with limited time. Getting more users to use it would be however a motivating factor :)

As you may have noticed, I got stuck with my pull request for update 4 on the main repository two months ago, requesting me to break it up into small peaces, which was just too much hassle for me.

What I try to say with this is, that while being the only contributor, I also need some freedom to change things where I think necessary without the reviewer being too picky about how I deliver it.

So, yes I would like to get involved in the main repository, but I need to be able to merge all my current changes from the last 3 months into master, because I don't want to maintain 2 different branches.

Let me know what you think about this and get going? ;)

Cheers.

**Comment by wokhan** [comment]
Hi @harrwiss,
I won't be lecturing anyone here saying how things should be merged or not (I've not been helpful enough the last 4 or 5 years to allow myself to do so 😁).
But... I think you should be able to merge everything at once through a single MR (I might take some time to review all of it if needed, so that @AtlasHackert is not the only one handling all the review) and then I'd ask you to comply with what he required, i.e. creating feature-scoped branches to ease MR reviews, which is still something valuable even if, well, this is only a 2 or 3 person's project....

Would you both agree with this?

Btw, regarding "rewards", when donations come, I split them amongst actually involved people. Don't expect anything that exciting (a few dollars a year... better than nothing I guess... 😁), but it could be an additional small motivation to join the main repo 😉

Thanks

**Comment by harrwiss** [comment]
Hi @wokhan , thanks for your quick response.
Sounds reasonable to me 👍

**Comment by leeuniverse** [comment]
Thanks harrwiss...

I'll go ahead and try it out.

**Comment by leeuniverse** [comment]
Okay, bad news....  @wokhan and @harrwiss WFN is actually WORSE than it was with the last update.
Sadly, this program is unusable, and I'm going to have to use SimpleWall instead.
However, notify me if you all do a major update, cause like I said, this is the best program, I WANT to use it.  We seriously have to fix these bugs.

The latest update has these new and continued problems.

1. Taskbar Icon will no longer show most of the time, and really ever save a brief moment for some reason soon after usage, but it didn't even show at start.

2. There is no icon in Task Manager "Startup" (there was before if I remember right).  Don't know where the startup icon is hiding to have disabled it.

3. The Notifications are even WAY worse than before.  In fact, they are so bad that on a system restart, and opening my main browser (which I had allowed a million times already), WFN did what MS Outlook did previously, had to restart my system and this was the final straw, had to also Reset Windows Defender to default (hope that didn't screw anything up).

4. So, the notification is doing two things I think.  First, it doesn't recognize that a program with the same ports, ip, etc. is already waiting for approval, so it creates a new notification on each try.  At first, I thought that was the problem, but the program also is later on after you've approved the same program, IP's, Ports, etc. it just notifies you yet again for the same EXACT connections.  So, something is wrong here in two respects.

I've attached 3 screenshots, examples of this.  One with a 3rd Party Anti-Virus program (but it does it with Browsers etc., so the problem is not unique to my anti-virus) and two with Windows Services.
Note, I think I mentioned above already, we also really need a setting in the program which we can allow all Windows OS connections (and then we can choose after to change the rules of some if we want).

![programsrepeating](https://user-images.githubusercontent.com/9868451/74386286-cbfb8f80-4db2-11ea-83cd-2ed2a8c9cb45.png)
![programsrepeating2](https://user-images.githubusercontent.com/9868451/74386287-cc942600-4db2-11ea-99e9-bac74dd11264.png)
![programsrepeating3](https://user-images.githubusercontent.com/9868451/74386289-cd2cbc80-4db2-11ea-831e-2569af749431.png)

Thanks guys....  Hope you can take some time to finally get this program worthy.
Hate that you've made such a quality product and it's falling to the wayside and not having the usage it should.

**Comment by harrwiss** [comment]
Hi @leeuniverse,
thanks for taking your time to check it out eventhough with bad news 🤔 

> Taskbar Icon will no longer show most of the time, and really ever save a brief moment for some reason soon after usage, but it didn't even show at start.
Think you relate to the Notifier icon here, you're correct there was a change to minimize the notifier to the windows notification area:
![image](https://user-images.githubusercontent.com/33778373/74478983-aeabeb80-4eae-11ea-8900-5f9426717794.png)
In your case, the icon is not visible - this can be changed in Windows Settings for the Taskbar e.g. type "Select which icons appear on the taskbar". When click on ^ here it should be visible given that it was activated.
![image](https://user-images.githubusercontent.com/33778373/74481100-5d9df680-4eb2-11ea-98f7-0d5af3f43f04.png)

> There is no icon in Task Manager "Startup" (there was before if I remember right). Don't know where the startup icon is hiding to have disabled it.

Think we never had one i.e. since we're not using an installer, a Startup icon is not be created automatically. 

> The Notifications are even WAY worse than before. In fact, they are so bad that on a system restart, and opening my main browser "360Chrome" (which I had allowed a million times already), WFN did what MS Outlook did previously, which is totally crash and no amount of clicking of the Notification bar to Allow would make it go away. Had to restart my system and this was the final straw, had to also Reset Windows Defender to default (hope that didn't screw anything up).

Not sure what you mean by totally crash - WFN, Windows? To disable WFN it should be enough to start WFN.exe and in Firewall Settings set "Block silently". Resetting the Windows Firewall will remove all manually created rules. 
Could you maybe provide a screenshot of the "Settings" and "Options" used?

> So, the notification is doing two things I think. First, it doesn't recognize that a program with the same ports, ip, etc. is already waiting for approval, so it creates a new notification on each try. At first, I thought that was the problem, but the program also is later on after you've approved the same program, IP's, Ports, etc. it just notifies you yet again for the same EXACT connections. So, something is wrong here in two respects.

I noticed the following in the screenshots:
There are many duplicates with profile "Public" but I haven't seen duplicates without a profile selected. Was there a specific reason to select the "Public" profile specifically? 
Personally I never select a specific profile in the advanced options (default) this way you can have one Allow/Block rule for all profiles.
See also here for some recommendations: [FAQ](https://github.com/harrwiss/WFN/wiki/FAQ)

Cheers.

**Comment by leeuniverse** [comment]
Hey @harrwiss ...

1. LOL, sorry I wasn't clear enough, but WNF "was" in fact set to show icons on the Taskbar, so no it wasn't just hidden behind the arrow.

2. Sorry again wasn't clear enough.  The Notification box simply froze there, wouldn't move, couldn't Allow etc.
I set WNF to show Notifications for ALL Zones In and Out...

3. I would always use the "default" settings in Notification over and over again, and only after that failing to stop the same pop-up, ports, IP, etc., I would then select ALL the options to try and see if that would fix it.
Oh wait, are you talking about the Profiles in Advanced Settings.  Like I mentioned above, I set them ALL to block all and to notify all both in and out.  Maybe that's been my problem, doing that isn't working right or something?  Cause, my interpretation of those settings is to make sure I'm controlling all connections in all zones, but I'm guessing that's not working right?

So, I'm assuming you're actually using this firewall, and you don't have my problems with the default profile settings?  However, the default won't show/block all connections right, so how do we do that without the bugs I've been describing?

**Comment by harrwiss** [comment]
Hello @leeuniverse 

Of course I'm using it myself :laughing: ... but sadly my first experience with it few months ago was not so much different to what you describe. But then I gave it another chance and tried to better understand how it works and adapting my mindset ... finally got to like it and slightly changed the way I used it :smirk:

>Notifier icon not being shown in the windows tray 

This could mean:
    
* Notifier was not triggered by a block event yet e.g. it only appears for connections which have neither a block or allow rule defined
* It crashed during startup - in this case you should see a related windows error report (Event viewer > Windows Logs > Appplication)

>The Notification box simply froze there, wouldn't move, couldn't Allow etc. Had to shut down the computer to fix it, trying to End Task the program etc. wouldn't do anything. I set WNF to block and show Notifications for ALL Zones In and Out...

* Could be caused by too many connections being blocked and processed making it unresponsive. If there are constantly new block events being created then even after killing Notifier in the TaskManager a new instance would be started immediately. To stop this from happening (without restarting) you should set "Block silently" on all profiles in WFN _firewall settings_; then go to WFN _security log_ and check which application tries to open so many connections.

>I would always use the "default" settings in Notification over and over again, and only after that failing to stop the same pop-up, ports, IP, etc., I would then select ALL the options to try and see if that would fix it.

* well, if you select all options, it does the opposite of what you want :smirk: What you should try then is, to disable all rule options except for Appllication path i.e. the less options you've select, the more global the rule is.

>Like I mentioned above, I set them ALL to block all and to notify all both in and out. Maybe that's been my problem, doing that isn't working right or something? Cause, my interpretation of those settings is to make sure I'm controlling all connections in all zones, but I'm guessing that's not working right?

You're refering to the profiles in WFN > _firewall settings_ I assume. They should work and your interpretation that ALL applies to all profiles appears correct. Just remember that this setting is only controlling whether you want to be prompted by opening a Notifier window for every blocked connection. In the Notifier window (Adv. rule settings) specifies to which profile the rule applies (disabled = all).

**Comment by harrwiss** [comment]
In summary I can now imagine, that the first-time experience for users is often not the best because:

* per default the windows firewall blocks incoming connections silently and allows most outgoing connections without prompting.
* when activating "Block and prompt" in WFN firewall settings, all outgoing connections become blocked by default and therefore a whole lot of notifications are being triggered. 
* today nearly all applications/services and even browsers try to make several connections to the internet in the background (you can see active ones in WFN under _connections_ or the audit trail in the _security log_). Some apps also miss-behave completey trying in a frenzy to re-establish a connection constantly.
* given the sudden vast amount of notifications to be handled a user becomes overwhelmed and starts clicking immediately to create rules without too much thinking (happend to me at least :smirk:)
* even Notifier itself starts to struggle being bombarded with too many block events in a short amount of time rendering it temporarely unresponsive. If however Notifier totaly crashes for whatever reason a related windows error report would be created (Event viewer > Windows Logs > Appplication). Less serious errors however are logged in the logfile located in the installation directory.

My feeling is, that the first-time user experience is primarely affected by how Notifier comes into play with all connections blocked. After the vast amount of Allow/Block rules have been created to ones liking things calm down significantly (my experience after all). I admit that there are still some issues with the rule detection (e.g. duplicate rules being created as a result in certain cases) but I don't think thats the root problem as long as we have not solved problems like freezing/crashing the Notifier on first use.

So in my opinion, the Notifier should no longer be started based on Windows security log events, because currently it is not predictable when and how often blocking events occur and further makes it difficult to optimize rule detection, resource consumption, error handling (when crashing), enabling/disabling prompting, UAC prompting, etc.

@wokhan / @AtlasHackert: This is of course only my point of view this far - anything to add?

However, I already started and use a working prototype which can be started stand-alone and polls the security log instead, but it needs still some effort on the ui and config side to publish it.

**Comment by leeuniverse** [comment]
BTW, just an FYI, I've been using 3rd Party Firewalls with Notifications/Allow/Disallow for some 30 years now, and I set them to Show ALL etc., so I'm well experienced with getting a lot of notifications initially and thus having to set the Allow/Disallow for programs.

WFN, however, has went psycho in that dept.  LOL  :)

**Comment by kekukui** [comment]
> **I have been looking for this program for YEARS...**

Same here... I have tested every Windows firewall (>50 apps) and I think all are dysfunctional in some way.  Every decent Windows firewall has been sold to some other company and converted into a bloated antivirus/security suite.  And all of the free firewalls which remain have a bad user interface, a confusing or messy connection request dialog, or they lack some important features.

> **Marketing has been your problem**

WFN has the potential to become a very popular Windows firewall, but some bugs need to be fixed before seeking more publicity: If a software project gets a bad reputation at the beginning of  development, many users will not return and try new versions later, when the major bugs are gone.  As Orson Welles said: _**"We will sell no wine before its time."**_  But as Guy Kawasaki said, _**"If you are not  embarrassed or ashamed of your first version, then you are shipping too late."**_  😉

> **We are so close from a finished product**

Yes, I agree... only a few bugs stand in the way.  At some point in the near future I expect that WFN  will be ready for the general public and then bloggers will review it, and more users & developers will  come aboard.

> **Documentation? That might be a bit harder, not that I couldn't do it**

If time permits, I might be able to contribute to the documentation occasionally -- but if the user interface is intuitive, you probably dont need a lot of documentation.  When the project is ready, you can get a free domain name that resolves to your free Github hosting site (wokhansoft.github.io).

https://github.blog/2013-04-05-new-github-pages-domain-github-io/
https://pages.github.com/
https://medium.com/swlh/how-to-host-your-website-on-github-pages-for-free-3302b0fe8956

> **"SimpleWall" has been working really well, but I'm looking forward to using WFN, it's definitely far  superior if we can get rid of the bugs.**

Simplewall is not bad for a lightweight desktop firewall, but sometimes maybe too simple, and I also would prefer WFN.

> **I tested "Free Firewall" and it's somewhat okay, but I think it's actually doing the SAME thing with some programs as yours with the "million notifications" BUG for the same connection/app**

If there are many alerts in the queue waiting for user approval, and one specific connection is  allowed, all duplicate alerts should be removed from the alert queue so the user does not have to answer the same question multiple times.  Many firewalls fail to handle this correctly.  But others fail in the opposite way: you miss some alerts because you cannot filter by port or subdomain or wildcards.  This is why I cannot find a desktop firewall that meets my needs.

> **My feeling is, that the first-time user experience is primarely affected by how Notifier comes into  play with all connections blocked.** 

Yes, exactly.  That is why I like the design of Little Snitch.  Sadly its only for Mac - but at least WFN has a similar network map.  This is a nice feature.

I prefer to create a rule for every connection, but you could have a preset which allows all Windows system traffic for people who only want to configure rules for desktop apps.  The '**Private WinTen**' firewall has a unique approach, with presets for blocking Windows telemetry:

![image](https://user-images.githubusercontent.com/20834957/76427493-c0d05b00-6382-11ea-91f4-244e4142e957.png)

https://github.com/DavidXanatos/priv10
https://www.ghacks.net/2018/12/26/private-winten-new-windows-10-privacy-tweaker-with-firewall/

(This app is also somewhat unstable, beta quality... but it does have some interesting features.)

**Comment by wokhan** [comment]
Hi, thanks for these.

A few years ago, WFN totaled more than 200k downloads (being on MajorGeeks, Download.com and some others) and was often mentioned in security forums, which is, IMO, already something.
Now that we are back to make it the app it should have been in the first place, I'm confident it can regain users (as in _power users_) attention. I don't want it to be a software for non power-users as I have no time to support them... even if the goal is to make it as easy to use and as user friendly as possible (which we'll achieve, believe me).
So... stay tuned, and I'm sure you'll be pleased by what's coming ;)
Regarding the website, I already made one (which is indeed a bit empty) at https://www.wokhan.com, a smart move could be to host the documentation on github, as you said, maybe displaying it on the main website using a markdown renderer (I don't like the idea of having multiple entry points for the same thing). That way the documentation would also be a community effort....

Thank you for your future help as a tester and documentation writer then ;-)

**Comment by harrwiss** [comment]
I sencond that:

>  I'm confident it can regain users (as in power users) attention. I don't want it to be a software for non power-users as I have no time to support them... even if the goal is to make it as easy to use and as user friendly as possible (which we'll achieve, believe me).

And @kekukui, thanks for the nice analysis and info provided 👍
